perceval/simulators/simulator_factory.py

```diff
@@ -137,5 +137,6 @@ def build(circuit: ACircuit | Processor | list,
 
         if convert_to_circuit:
             circuit = _unitary_components_to_circuit(circuit, m)
-        simulator.set_circuit(circuit)
+        # m can only be retrieved from a Processor (and useful only in that case)
+        simulator.set_circuit(circuit, m if m else None)
```

Comment:
I think we need an non-linear optics circuit class that will share the same basic interface with the Circuit class in order to give a circuit object to the set_circuit method that will have the m info in it instead of changing all the calls.